src/components/Image.tsx

```diff
@@ -0,0 +1,200 @@
+    // remove the rotate option
+    delete options.rotate;
+  }
+
+  // if sharpen is an option, make sure it is in between 1 and 10
+  if (options?.sharpen && (options.sharpen < 1 || options.sharpen > 10)) {
+    console.warn("Sharpen prop must be a number between 1 and 10");
+    // remove the sharpen option
+    delete options.sharpen;
+  }
+
+  // if quality is an option, make sure it is in between 1 and 100
+  if (options?.quality && (options.quality < 1 || options.quality > 100)) {
+    console.warn("Quality prop must be a number between 1 and 100");
+    // remove the quality option
+    delete options.quality;
+  }
+
+  // format trim into a string of the form 20;30;20;0 where the number is 0 if not specified
+  let trim = "";
+  if (options?.trim) {
+    const { top, right, bottom, left } = options.trim;
+    trim = `${top ?? 0};${right ?? 0};${bottom ?? 0};${left ?? 0}`;
+  }
+
+  // check if background is a valid CSS color value
+  if (options?.background && !COLOR_REGEX.test(options.background)) {
+    console.warn("Background prop must be a valid CSS color value");
+    // remove the background option
+    delete options.background;
+  }
+
+  let baseUrl = imageUrl;
+  const uuid = getImageUUID(baseUrl);
+
+  // change the a.mktgcdn.com in the url to dyn.mktgcdn.com and drop the dimensions and extension from the end of the url
+  baseUrl = baseUrl.replace("a.mktgcdn.com", "dyn.mktgcdn.com");
+  // remove the dimensions and extension from the end of the url
+  baseUrl = baseUrl.replace(/\/\d+x\d+\.\w+$/, "");
+
+  // built the options string
+  if (options) {
+    const {
+      width,
+      height,
+      quality,
+      // format,
+      fit,
+      background,
+      // gravity,
+      // blur,
+      brightness,
+      compression,
+      contrast,
+      dpr,
+      rotate,
+      // trim,
+      sharpen,
+      blur,
+    } = options;
+
+    const optionsArray = [
+      width && `width=${width}`,
+      height && `height=${height}`,
+      quality && `quality=${quality}`,
+      // format && `format=${format}`,
+      fit && `fit=${fit}`,
+      background && `background=${background}`,
+      // gravity && `gravity=${gravity}`,
+      // blur && `blur=${blur}`,
+      brightness && `brightness=${brightness}`,
+      compression && `compression=${compression}`,
+      contrast && `contrast=${contrast}`,
+      dpr && `dpr=${dpr}`,
+      rotate && `rotate=${rotate}`,
+      trim && `trim=${trim}`,
+      sharpen && `sharpen=${sharpen}`,
+      blur && `blur=${blur}`,
+    ].filter(Boolean);
+
+    if (optionsArray.length > 0) {
+      baseUrl = `${baseUrl}/${optionsArray.join(",")}`;
+    }
+  }
+  return baseUrl;
+};
+
+export const getImageUUID = (url: string) => {
+  const matches = url.match(MKTGCDN_URL_REGEX);
+
+  if (!matches?.groups?.uuid) {
+    console.error(`Invalid image url: ${url}.`);
+    return "";
+  }
+
+  return matches.groups.uuid;
+};
+
+// export const Image = ({
```

Comment:
so is this module not exporting a component?

Reply:
I can move this function into a util. I want to build out a robust component but for now I am just using the one function in my StyledComponents module

Reply:
The buildImageUrl function that is

Reply:
I think its okay for now, but I find this difficult to read. Let's leave for now